src/Core/Resolvers/MySqlQueryBuilder.cs

```diff
@@ -177,20 +175,47 @@ public override string BuildForeignKeyInfoQuery(int numberOfParameters)
         }
 
         /// <summary>
-        /// Makes the query segments to store PK during an update
+        /// Makes the query segments to store PK during an update. For each of the constructed segments, we do not include fields which are
+        /// read-only because read-only fields cannot be included in an update statement as their value cannot be updated. And consequently,
+        /// they cannot be included in the subsequent select statement as well.
         /// </summary>
-        private (string, string, string) MakeStoreUpdatePK(List<string> columns,
-                                                           List<LabelledColumn> outputColumns)
+        /// <param name="structure">Query structure of the update/upsert query.</param>
+        /// <param name="outputColumns">List of columns to be returned.</param>
+        /// <returns>A tuple of 3 strings where:
+        /// 1. The first string is for the set clause: to create local variables to store the updatable columns.
+        /// 2. The second string is for the update clause: to fetch the values of the updatable columns to local variables.
+        /// 3. The third string is for the select clause: to select local variables and mapping to original column name.
+        /// </returns>
+        private (string, string, string) MakeQuerySegmentsForUpdate(BaseSqlQueryStructure structure, List<LabelledColumn> outputColumns)
         {
-            // Create local variables to store the columns
-            string sets = String.Join(";\n", columns.Select((x, index) => $"SET {"@LU_" + index.ToString()} := 0"));
-
-            // Fetch the value to local variables
-            string updates = String.Join(", ", columns.Select((x, index) =>
-                $"{QuoteIdentifier(x)} = (SELECT {"@LU_" + index.ToString()} := {QuoteIdentifier(x)})"));
-
-            // Select local variables and mapping to original column name
-            string select = String.Join(", ", outputColumns.Select((x, index) => $"{"@LU_" + index.ToString()} AS {QuoteIdentifier(x.Label)}"));
+            SourceDefinition sourceDefinition = structure.GetUnderlyingSourceDefinition();
+            List<string> columns = structure.AllColumns();
```

Comment:
This may be an issue where you are resolving all columns for the subsequent SELECT. The requestor may not have read permissions for all columns or the requestor may have listed only a subset of columns in the GraphQL selection set or REST query string ($select).

Reply:
Not a new change that is introduced in this PR. As Ani mentioned: https://github.com/Azure/data-api-builder/pull/1596#issuecomment-1712358895, it is a TODO with @severussundar.